### ISSUE TITLE
Typescript fails to compile with axios-retry

### ISSUE THREAD

**Issue by you-fail-me** [softonic/axios-retry#41]
axios-retry version: v3.0.2

typescript version 2.8.1

Fails to compile, getting
```
node_modules/axios-retry/index.d.ts(27,3): error TS7020: Call signature, which lacks return-type annotation, implicitly has an 'any' return type.
```

When I hover over that line in the source code I get the same error. What is the reason? How can it be fixed? Don't mind making a PR if someone explains a bit on that

**Comment by abelino** [comment]
Are you running into any runtime errors? I am running into `TypeError: axios_retry_1.default is not a function` when importing as the docs suggest:
`import axiosRetry from 'axios-retry'`

switching to `import * as axiosRetry from 'axios-retry'` fixes the runtime error but obviously results in a type error

**Comment by you-fail-me** [comment]
The error you're getting looks like something misconfigured with typescript. I don't use synthetic default exports and do `import * as axiosRetry from 'axios-retry'` and it works for me. I'd check tsconfig for possible reasons

**Comment by ignacio-blas** [comment]
Hi, I has same error with TypeScript and axios-retry 3.1.0.

In index.d.ts the code in line 33-38 contains this interface:

    export interface IAxiosRetry {
      (
        axios: axios.AxiosStatic | axios.AxiosInstance,
        axiosRetryConfig?: IAxiosRetryConfig
      )
    }

This interface defines its params types, but not its return type. If you have a TypeScript configuration with strict type checking option "--noImplicitReturns", this is a compilation error.

`axiosError()` don't return anything, so is enough with explicitly set return type as `void` in index.d.ts:

    export interface IAxiosRetry {
      (
        axios: axios.AxiosStatic | axios.AxiosInstance,
        axiosRetryConfig?: IAxiosRetryConfig
      ): void
    }

**Comment by nishimura-yuji** [comment]
Hi, I has similar error with TypeScript and axios-retry 3.1.2.

```
Cannot invoke an expression whose type lacks a call signature. Type 'typeof import("/Users/nishimura.yuji/work/axios/node_modules/axios-retry/index")' has no compatible call signatures.
```

**Comment by c1moore** [comment]
The issue is here
```
module.exports = require('./lib/index').default;
```

Instead of exporting the default export, you should just export the index file:

**Comment by LJJCherry** [comment]
> Hi, I has similar error with TypeScript and axios-retry 3.1.2.
> 
> ```
> Cannot invoke an expression whose type lacks a call signature. Type 'typeof import("/Users/nishimura.yuji/work/axios/node_modules/axios-retry/index")' has no compatible call signatures.
> ```

I met the same problem, how to fix

**Comment by c1moore** [comment]
The type fix was released in version 3.1.3.  Try upgrading.
